src/sigmatcher/analysis.py

```diff
@@ -219,28 +178,75 @@ def analyze(self, unpacked_path: Path, results: Dict["Analyzer", Union[Result, E
         parent_class_result.matched_methods.append(matched_method)
         return matched_method
 
+    def get_dependencies(self) -> Set[str]:
+        return self.definition.get_dependencies() | {self.parent.name}
+
     @property
     def name(self) -> str:
         return f"{self.parent.name}.methods.{self.definition.name}"
 
 
-def analyze(definitions: Definitions, unpacked_path: Path) -> None:
-    analyzers: List[Analyzer] = []
+@dataclasses.dataclass(frozen=True)
+class ExportAnalyzer(Analyzer):
+    definition: ExportDefinition
+    parent: ClassAnalyzer
+
+    def analyze(self, unpacked_path: Path, results: Dict[str, Union[Result, Exception, None]]) -> MatchedExport:
+        parent_class_result = results[self.parent.name]
+        assert isinstance(parent_class_result, MatchedClass)
+
+        raw_class = parent_class_result.smali_file.read_text()
+        signature = self.definition.signatures[0].resolve_macros(results)
+        captured_names = set(signature.capture(raw_class))
+        check_match_count(captured_names, self.name)
+        export_value = next(iter(captured_names))
```

Comment:
Maybe allow using a special named capture group to choose which one is the result?  
This will allow using references in regex patterns (`/(a)\1/ == /aa/`).

Reply:
How useful do think using backrefrences will be?
I'm asking because I think it would be annoying to write something like `(?P<match>A00:LX/001;)` in every definition.
Also if I implement it right now it will use python's syntax for named caputre (`?P<match>)`) but in the future, I might change the regex backend for fields, exports and methods to ripgrep, which mean the syntax will change.

Reply:
Well apparently ripgrep does support the same syntax as python, so you can ignore the second part.

Reply:
I use backreferences for tracking registers extensively:
```
    const-string	v1, "Hello World!"
    invoke-virtual {v0, v1}, LX/001;->A00(Ljava/lang/String;)V
```
can be best matched with:
```
    const-string	(v\d+), "Hello World!"
    invoke-virtual \{v\d+, \1\}, (?P<match>LX/001;)->A00\(Ljava/lang/String;\)V
```

I do agree that it'll be awful to always require a named capture - maybe default to the first one if it there isn't a named one?